speech/grpc/src/main/java/com/google/cloud/speech/grpc/demos/AsyncRecognizeClient.java

```diff
@@ -0,0 +1,248 @@
+        .negotiationType(NegotiationType.TLS)
+        .intercept(new ClientAuthInterceptor(creds, Executors.newSingleThreadExecutor()))
+        .build();
+    stub = SpeechGrpc.newBlockingStub(channel);
+    statusStub = OperationsGrpc.newBlockingStub(channel);
+
+    logger.info("Created stub for " + host + ":" + port);
+  }
+
+  private RecognitionAudio createRecognitionAudio() throws IOException {
+    return RecognitionAudioFactory.createRecognitionAudio(this.input);
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /** Send an async-recognize request to server. */
+  public void recognize() {
+    RecognitionAudio audio;
+    try {
+      audio = createRecognitionAudio();
+    } catch (IOException e) {
+      logger.log(Level.WARNING, "Failed to read audio uri input: " + input);
+      return;
+    }
+    logger.info("Sending " + audio.getContent().size() + " bytes from audio uri input: " + input);
+    RecognitionConfig config = RecognitionConfig.newBuilder()
+        .setEncoding(AudioEncoding.LINEAR16)
+        .setSampleRate(samplingRate)
+        .build();
+    AsyncRecognizeRequest request = AsyncRecognizeRequest.newBuilder()
+        .setConfig(config)
+        .setAudio(audio)
+        .build();
+
+    Operation operation;
+    Operation status;
+    try {
+      operation = stub.asyncRecognize(request);
+
+      //Print the long running operation handle
+      logger.log(Level.INFO, String.format("Operation handle: %s, URI: %s", operation.getName(),
+            input.toString()));
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
+    }
+
+    while (true) {
+      try {
+        logger.log(Level.INFO, "Waiting 2s for operation, {0} processing...", operation.getName());
+        Thread.sleep(2000);
+        GetOperationRequest operationReq = GetOperationRequest.newBuilder()
+            .setName(operation.getName())
+            .build();
+        status = statusStub.getOperation(
+            GetOperationRequest.newBuilder()
+                .setName(operation.getName())
+                .build()
+                );
+
+        if (status.getDone()) {
+          break;
+        }
+      } catch (Exception ex) {
+        logger.log(Level.WARNING, ex.getMessage());
+      }
+    }
+
+    try {
+      AsyncRecognizeResponse asyncRes = status.getResponse().unpack(AsyncRecognizeResponse.class);
+
+      logger.info("Received response: " + asyncRes);
+    } catch (com.google.protobuf.InvalidProtocolBufferException ex) {
+      logger.log(Level.WARNING, "Unpack error, {0}",ex.getMessage());
+    }
+  }
+
+  public static void main(String[] args) throws Exception {
+
+    String audioFile = "";
+    String host = "speech.googleapis.com";
+    Integer port = 443;
+    Integer sampling = 16000;
+
+    CommandLineParser parser = new DefaultParser();
```

Comment:
To keep things simpler, Java samples usually use "properties" for these. See the bookshelf sample: https://github.com/GoogleCloudPlatform/getting-started-java/blob/master/bookshelf/4-auth/pom.xml#L28

Hmm... Seems like that may be for servers, only, though... https://github.com/GoogleCloudPlatform/getting-started-java/blob/48bfbe9530b313ffec17b32edf959a38010bae3a/bookshelf/4-auth/src/main/java/com/example/managedvms/gettingstartedjava/basicactions/ListBookServlet.java#L50